packages/terra-date-picker/src/react-datepicker/index.jsx

```diff
@@ -396,6 +399,29 @@ class DatePicker extends React.Component {
     }
   }
 
+  handleOnDayMouseDown() {
+    if (this.props.inline) {
+      // prevents focus border on pre-selected day on mouseDown when calendar is inline.
+      this.setState({ preSelection : null })
+    }
+  }
+
+  boundedPreSelection() {
+    const defaultPreSelection = this.getPreSelection()
+    const minDate = getEffectiveMinDate(this.props)
+    const maxDate = getEffectiveMaxDate(this.props)
+    return minDate && isBefore(defaultPreSelection, minDate) ? minDate
+    : maxDate && isAfter(defaultPreSelection, maxDate) ? maxDate
+      : defaultPreSelection;
+  }
+
+  handleMonthBlur() {
+    if (this.props.inline) {
+      // resets previous selected day to selected or current day from previous focused day ( non-selected ) when calendar is inline.
+      this.setState({ preSelection : this.props.selected ? newDate(this.props.selected) : this.boundedPreSelection() })
```

Comment:
I don't know that we should reset the preSelection when it loses focus. Wouldn't the user lose the preselected day when it regains focus?

Would it fix the focus issue if we just set `this.state.isCalendarKeyboardFocused` to false when it loses focus instead of resetting the preselected day?

Reply:
we are doing  `this.state.isCalendarKeyboardFocused` and `this.state.isCalendarOpenedViaKeyboard` = false in on the [blur event of month in calendar.jsx](https://github.com/cerner/terra-framework/blob/7b998f44f432b8a12db6133688e730b8cc60ed02/packages/terra-date-picker/src/react-datepicker/calendar.jsx#L365). 
 
Since Calendar-filter will always be visible we would need focus to be set on the selected day / current day / 1st day of month when Calendar-filter regains focus. If we don't reset the pre-selected day to valid day on blur, focus will shift to un-selected day will be get highlighted when it regains focus. 

![pre-selected-focus](https://user-images.githubusercontent.com/12869809/87660400-9d784b00-c77c-11ea-89a2-c329f81bd0a4.gif)

in the above gif we can see that on focusing back focus moves to previously navigated day instead of focusing selected date.